v2/tools/asoctl/internal/crd/cleaner.go

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package crd
+
+import (
+	"context"
+	"regexp"
+	"time"
+
+	"github.com/Azure/azure-service-operator/v2/internal/logging"
+	"github.com/pkg/errors"
+
+	apiextensions "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Cleaner struct {
+	apiExtensionsClient apiextensionsclient.CustomResourceDefinitionInterface
+	client              client.Client
+	migrationBackoff    wait.Backoff
+	dryRun              bool
+}
+
+func NewCleaner(apiExtensionsClient apiextensionsclient.CustomResourceDefinitionInterface, client client.Client, dryRun bool) *Cleaner {
+	migrationBackoff := wait.Backoff{ // TODO: Still need to see if we want exponential backoff or linear is fine
```

Comment:
Theoretically linear is useful, but it's just as easy to use Exponential and that has none of the untoward side effects.

Reply:
Best practice is to include some jitter as well.

Reply:
`retry.OnError` uses exponential backoff under the hood.
https://github.com/kubernetes/client-go/blob/master/util/retry/util.go#L50